crates/biome_dependency_graph/src/dependency_graph.rs

```diff
@@ -61,20 +67,12 @@ impl DependencyGraph {
     /// `added_or_updated_paths` and `removed_paths`. Manifests are expected to
     /// be resolved through the `project_layout`. As such, the `project_layout`
     /// must have been updated before calling this method.
-    ///
-    /// `get_js_syntax_for_path` is a callback that may be called for any of the
-    /// files given in `added_or_updated_paths`, and it should return the syntax
-    /// root for each of them. If a file is already removed, or is inaccessible,
-    /// by the time `get_js_syntax_for_path` is called for it, `None` must be
-    /// returned. Error reporting, if necessary, should be handled by the
-    /// workspace server instead.
     pub fn update_graph_for_js_paths(
         &self,
         fs: &dyn FileSystem,
         project_layout: &ProjectLayout,
-        added_or_updated_paths: &[BiomePath],
+        added_or_updated_paths: &[(&BiomePath, Option<AnyJsRoot>)],
         removed_paths: &[BiomePath],
-        get_js_syntax_for_path: impl Fn(&Utf8Path) -> Option<AnyJsRoot>,
```

Comment:
Removing the callback in favour of adding the `AnyJsRoot` directly to the `added_or_updated_paths` is a small refactor that's actually unrelated to most of the PR. I did this when I was still toying with the idea of creating a type info model _next to_ the dependency graph. But I expect it improves performance, as each invocation of the callback used to pin the `documents` map in the workspace server, so I decided to keep it.